### PR TITLE
GH1692: Log more diagnostics from Roslyn.

### LINKED ISSUE #1692: Log script compilation warnings and other diagnostics
Should be doable after #1645 is merged

### PR DESCRIPTION
**Pull request by mholo65**
- Fixes #1692

### CODE REVIEW

File: src/Cake/Scripting/Roslyn/RoslynScriptSession.cs [hunk 1]
```diff
@@ -96,10 +96,30 @@ public void Execute(Script script)
             var compilation = roslynScript.GetCompilation();
             var diagnostics = compilation.GetDiagnostics();
 
-            if (diagnostics.Any(d => d.Severity == DiagnosticSeverity.Error))
+            var errors = new List<Diagnostic>();
```
Comment (gep13) [inline src/Cake/Scripting/Roslyn/RoslynScriptSession.cs]: @mholo65 am I missing something here, or are you never adding anything to `errors`?

Comment (mholo65) [reply src/Cake/Scripting/Roslyn/RoslynScriptSession.cs]: Oh, minor issue happened during refactoring. Nice catch!

**Comment by devlead** [comment]
@mholo65 could you address feedback and rebase against latest develop?

**Comment by devlead** [comment]
@mholo65 your changes have been merged, thanks for your contribution 👍

**Pull request merged**